pkg/controller/controlplanecerts/controlplanecerts_controller.go

```diff
@@ -361,6 +357,12 @@ func (r *ReconcileControlPlaneCerts) setCertsNotFoundCondition(cd *hivev1.Cluste
 	return true, r.Status().Update(context.TODO(), cd)
 }
 
+// defaultControlPlaneDomain will return the string representing the DNS entry
+// for the cluster's API endpoing
+func defaultControlPlaneDomain(cd *hivev1.ClusterDeployment) string {
+	return fmt.Sprintf("api.%s.%s", cd.Spec.ClusterName, cd.Spec.BaseDomain)
```

Comment:
I do not like that this is hard-coded. This will be a nightmare if the installer changes what the API endpoint is at some point.

Reply:
Agreed, the Status value was extracted from kubeconfig IIRC and should be resilient to installer changes, more commit details would help here.

Reply:
now using url parsing to pull out the hostname/domain